gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceConnector.java

```diff
@@ -80,11 +95,25 @@ public HttpEntity getAuthentication() throws RestApiConnectionException {
 
       return httpEntity;
     } catch (Exception e) {
-      throw new RestApiConnectionException("Failed to authenticate salesforce using user:" + userName + " and host:"
+      throw new RestApiConnectionException("Failed to authenticate salesforce host:"
           + host + "; error-" + e.getMessage(), e);
     }
   }
 
+  @Override
+  protected void addHeaders(HttpRequestBase httpRequest) {
+    if (this.accessToken != null) {
+      httpRequest.addHeader("Authorization", "Bearer " + this.accessToken);
```

Comment:
What about the original OAuth of the base method? If that is valid for Salesforce then we need to preserve that behavior.

Reply:
Although "Authorization: Bearer [access token]" is in RFC and Salesforce document, to prevent breaking existing integration, this header will only be used if access token comes from refresh token.